internal/driver/fetch_test.go

```diff
@@ -210,112 +210,299 @@ func TestFetchWithBase(t *testing.T) {
+		},
+		{
+			"diff_base and base both specified",
+			[]string{path + "cppbench.contention"},
+			[]string{path + "cppbench.contention"},
+			[]string{path + "cppbench.contention"},
+			false,
+			[]WantSample{},
+			"-base and -diff_base flags cannot both be specified",
 		},
 	}
 
 	for _, tc := range testcases {
-		t.Run(tc.desc, func(t *testing.T) {
-			pprofVariables = baseVars.makeCopy()
+		pprofVariables = baseVars.makeCopy()
 
-			base := make([]*string, len(tc.bases))
-			for i, s := range tc.bases {
-				base[i] = &s
-			}
+		base := make([]*string, len(tc.bases))
+		for i, s := range tc.bases {
+			base[i] = &s
+		}
 
-			f := testFlags{
-				stringLists: map[string][]*string{
-					"base": base,
-				},
-				bools: map[string]bool{
-					"normalize": tc.normalize,
-				},
-			}
-			f.args = tc.sources
+		diffBase := make([]*string, len(tc.diffBases))
+		for i, s := range tc.diffBases {
+			diffBase[i] = &s
+		}
 
-			o := setDefaults(&plugin.Options{
-				UI:      &proftest.TestUI{T: t, AllowRx: "Local symbolization failed|Some binary filenames not available"},
-				Flagset: f,
-			})
-			src, _, err := parseFlags(o)
+		f := testFlags{
+			stringLists: map[string][]*string{
+				"base":      base,
+				"diff_base": diffBase,
+			},
+			bools: map[string]bool{
+				"normalize": tc.normalize,
+			},
+		}
+		f.args = tc.sources
 
-			if err != nil {
-				t.Fatalf("%s: %v", tc.desc, err)
-			}
+		o := setDefaults(&plugin.Options{
+			UI:      &proftest.TestUI{T: t, AllowRx: "Local symbolization failed|Some binary filenames not available"},
+			Flagset: f,
+		})
+		src, _, err := parseFlags(o)
 
-			p, err := fetchProfiles(src, o)
-			pprofVariables = baseVars
-			if err != nil {
-				t.Fatal(err)
+		if tc.wantErrorMsg != "" {
+			if err == nil {
+				t.Errorf("%s: want error %q, got nil", tc.desc, tc.wantErrorMsg)
+				continue
 			}
-
-			if want, got := len(tc.expectedSamples), len(p.Sample); want != got {
-				t.Fatalf("want %d samples got %d", want, got)
+			gotErrMsg := err.Error()
+			if gotErrMsg != tc.wantErrorMsg {
+				t.Errorf("%s: want error %q, got error %q", tc.desc, tc.wantErrorMsg, gotErrMsg)
 			}
+			continue
+		}
+
+		if err != nil {
+			t.Errorf("%s: %v", tc.desc, err)
+			continue
+		}
 
-			if len(p.Sample) > 0 {
-				for i, sample := range p.Sample {
-					if want, got := len(tc.expectedSamples[i]), len(sample.Value); want != got {
-						t.Errorf("want %d values for sample %d, got %d", want, i, got)
-					}
-					for j, value := range sample.Value {
-						if want, got := tc.expectedSamples[i][j], value; want != got {
-							t.Errorf("want value of %d for value %d of sample %d, got %d", want, j, i, got)
-						}
-					}
+		p, err := fetchProfiles(src, o)
+
+		if err != nil {
+			t.Errorf("%s: %v", tc.desc, err)
+			continue
+		}
+
+		if want, got := len(tc.wantSamples), len(p.Sample); want != got {
+			t.Errorf("%s: want %d samples got %d", tc.desc, want, got)
+			continue
+		}
+
+		if len(p.Sample) > 0 {
```

Comment:
I think this check is redundant - if p.Sample is an empty slice the loop will immediately terminate anyway.

Reply:
Done